x-pack/plugins/fleet/public/applications/fleet/components/fleet_server_instructions/advanced_tab.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiSteps } from '@elastic/eui';
+import React from 'react';
+
+import { useAdvancedForm } from './hooks';
+
+import {
+  getAddFleetServerHostStep,
+  getSelectAgentPolicyStep,
+  getGenerateServiceTokenStep,
+  getSetDeploymentModeStep,
+  getInstallFleetServerStep,
+  getConfirmFleetServerConnectionStep,
+} from './steps';
+
+export const AdvancedTab: React.FunctionComponent = () => {
+  const {
+    eligibleFleetServerPolicies,
+    refreshEligibleFleetServerPolicies,
+    fleetServerPolicyId,
+    setFleetServerPolicyId,
+    isFleetServerReady,
+    serviceToken,
+    isLoadingServiceToken,
+    generateServiceToken,
+    fleetServerHostForm,
+    deploymentMode,
+    setDeploymentMode,
+  } = useAdvancedForm();
```

Comment:
I was a little on the fence about wrapping all of this data up into a single hook like this, and I could probably be convinced that it'd be better to simply compose multiple hooks in these "tab" files instead. The only real problem this solves is in `managed_instructions.tsx` where we import this same `useAdvancedForm` hook because we need almost all of this state again, so it made sense to simply ship the "composed" set of hooks all at once to prevent too much repetition between these two places. 

However, I could see the argument that our hooks should be a little more specific than this, which is very coupled to a given UI.